packages/dd-trace/src/appsec/index.js

```diff
@@ -121,8 +121,7 @@ function incomingHttpEndTranslator ({ req, res }) {
     payload[addresses.HTTP_INCOMING_COOKIES] = req.cookies
   }
 
-  // TODO: no need to analyze it if it was already done by the body-parser hook
-  if (req.query !== undefined && req.query !== null) {
+  if (req.query !== undefined && req.query !== null && typeof req.query === 'object') {
```

Comment:
look at how it's done just above, is it really needed to chcek for undefined then null then typeof ?
if typeof === object then you only need to chcek for null, because null is an object